bindings/twilio/sendgrid/sendgrid.go

```diff
@@ -180,6 +182,30 @@ func (sg *SendGrid) Invoke(ctx context.Context, req *bindings.InvokeRequest) (*b
 		bccAddress = mail.NewEmail("", req.Metadata["emailBcc"])
 	}
 
+	// Build email Dynamic Template Id, this is optional
+	templateID := ""
+	if sg.metadata.DynamicTemplateID != "" {
+		templateID = sg.metadata.DynamicTemplateID
+	}
+	if req.Metadata["dynamicTemplateId"] != "" {
+		templateID = req.Metadata["dynamicTemplateId"]
+	}
+
+	// Build email dynamic template, this is optional
+	var templateData map[string]interface{}
+	if sg.metadata.DynamicTemplateData != "" {
+		templateError := UnmarshalDynamicTemplateData(sg.metadata.DynamicTemplateData, &templateData)
```

Comment:
Because `sg.metadata` is parsed in the Init method, you should be able to do this un-marshaling only once and cache the result. This should improve perf quite a bit